src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -24,24 +27,31 @@ export const TransactionContract = ({
   let assets = []
   if (txDetails.sellAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Source))
   if (txDetails.buyAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Destination))
+  const translate = useTranslate()
   const isReceive = txDetails.tradeTx?.type === TransferType.Receive
   const interactsWithWithdrawMethod = txDetails.tx.data?.method === ContractMethod.Withdraw
   const isSend = txDetails.tradeTx?.type === TransferType.Send
   const i18n =
     isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const titlePrefix = translate(
+    txDetails.tx.data?.parser
+      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${i18n}`
+      : 'transactionRow.unknown',
+  )
+
+  const asset = useAppSelector(s => selectAssetById(s, (txDetails.tx.data as any).assetId! ?? ''))
```

Comment:
draft as i got a question about `TxMetadata` type in lib - at runtime it seems like we have `txDetails.tx.data.assetId` but not at compile time

Reply:
https://github.com/shapeshift/lib/blob/main/packages/chain-adapters/src/types.ts#L112 may need `assetId` added

Reply:
https://github.com/shapeshift/web/blob/develop/src/hooks/useTxDetails/useTxDetails.ts#L157 

this is a better place to figure the symbol

Reply:
We will need to use this already exposed type from unchained-client:

https://github.com/shapeshift/lib/blob/2b89bc53b99526361200c2b61febbe6a872182a7/packages/unchained-client/src/types.ts#L77-L89